### PR TITLE
clean up unparserfilternew to make it do less work that isn't printing

### PR DESCRIPTION
**Pull request by dwightguth**
**please review**

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
@radumereuta please review

@grosu please review changes to the tutorial

File: k-core/src/main/java/org/kframework/backend/unparser/AddBracketsFilter.java [hunk 1]
```diff
@@ -462,7 +462,7 @@ private boolean needsParentheses(Term inner, Term outer, Term leftCapture, Term
                     return true;
                 } else if (assoc == Associativity.RIGHT && !inversePriority) {
                     return true;
-                } else if (assoc == Associativity.LEFT && !inversePriority && !getAssociativity(inner, rightCapture, Associativity.RIGHT)) {
+                } else if (assoc == Associativity.LEFT && !inversePriority && !getAssociativity(inner, rightCapture, Associativity.LEFT)) {
```
Comment (radumereuta) [inline k-core/src/main/java/org/kframework/backend/unparser/AddBracketsFilter.java:465]: Was this a bug?

Comment (dwightguth) [inline k-core/src/main/java/org/kframework/backend/unparser/AddBracketsFilter.java:465]: Yes, unfortunately, when I was cleaning up this class earlier by making it use the associativity filters you created, I accidentally got this backwards.

**Comment by grosu** [comment]
lgtm, but i think @andreistefanescu should also take a look at kernelc.

**Comment by radumereuta** [comment]
code lgtm.

**Comment by andreistefanescu** [comment]
lgtm

**Pull request merged**